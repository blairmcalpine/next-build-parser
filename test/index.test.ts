import { expect, beforeEach, describe, vi, it } from 'vitest';
import { execute } from '../src/index.js';
import fs from 'node:fs';

const createArgv = (argv: string[]) => [
  process.argv[0],
  process.argv[1],
  ...argv,
];

describe('next-build-parser', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should parse the build file', () => {
    const result = execute(createArgv(['test/basic.txt']));
    expect(result).toMatchSnapshot();
  });
  it('should parse the build file and output based on the --output flag', () => {
    const result = execute(createArgv(['test/basic.txt', '--unit', 'B']));
    expect(result).toMatchSnapshot();
  });

  it('should write the output to a file if the --output flag is provided', () => {
    vi.spyOn(fs, 'writeFileSync');
    execute(createArgv(['test/basic.txt', '--output', 'output.json']));
    expect(fs.writeFileSync).toHaveBeenCalledWith(
      'output.json',
      expect.any(String),
    );
  });

  it('should parse the middleware output', () => {
    const result = execute(createArgv(['test/middleware.txt']));
    expect(result.middleware).toBe(0);
  });
});
