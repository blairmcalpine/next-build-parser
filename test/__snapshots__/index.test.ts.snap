// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`next-build-parser > should parse the build file 1`] = `
{
  "routes": [
    {
      "firstLoad": 125,
      "route": "/",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1.1,
    },
    {
      "firstLoad": 86.2,
      "route": "/_not-found",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 0.85,
    },
    {
      "firstLoad": 0,
      "route": "/api/trpc/[trpc]",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 0,
    },
    {
      "firstLoad": 140,
      "route": "/create",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 15.5,
    },
    {
      "firstLoad": 125,
      "route": "/popular",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1.1,
    },
    {
      "firstLoad": 132,
      "route": "/puzzle/[id]",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 8.15,
    },
    {
      "firstLoad": 0,
      "route": "/puzzle/[id]/opengraph-image",
      "routeType": "Edge",
      "routeTypeIcon": "ℇ",
      "size": 0,
    },
    {
      "firstLoad": 125,
      "route": "/recent",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1.1,
    },
  ],
  "sharedByAll": [
    {
      "chunk": "chunks/472-9100b5fcfec8f88c.js",
      "size": 30,
    },
    {
      "chunk": "chunks/fd9d1056-853251486b1823c6.js",
      "size": 53.3,
    },
    {
      "chunk": "chunks/main-app-842b2bfec3a562ae.js",
      "size": 0.23,
    },
    {
      "chunk": "chunks/webpack-b7a1470890cdd00b.js",
      "size": 1.79,
    },
  ],
  "sharedByAllTotal": 85.3,
  "unit": "kB",
}
`;

exports[`next-build-parser > should parse the build file and output based on the --output flag 1`] = `
{
  "routes": [
    {
      "firstLoad": 128000,
      "route": "/",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1126.4,
    },
    {
      "firstLoad": 88268.8,
      "route": "/_not-found",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 875,
    },
    {
      "firstLoad": 0,
      "route": "/api/trpc/[trpc]",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 0,
    },
    {
      "firstLoad": 143360,
      "route": "/create",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 15872,
    },
    {
      "firstLoad": 128000,
      "route": "/popular",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1126.4,
    },
    {
      "firstLoad": 135168,
      "route": "/puzzle/[id]",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 8345.6,
    },
    {
      "firstLoad": 0,
      "route": "/puzzle/[id]/opengraph-image",
      "routeType": "Edge",
      "routeTypeIcon": "ℇ",
      "size": 0,
    },
    {
      "firstLoad": 128000,
      "route": "/recent",
      "routeType": "Dynamic",
      "routeTypeIcon": "λ",
      "size": 1126.4,
    },
  ],
  "sharedByAll": [
    {
      "chunk": "chunks/472-9100b5fcfec8f88c.js",
      "size": 30720,
    },
    {
      "chunk": "chunks/fd9d1056-853251486b1823c6.js",
      "size": 54579.2,
    },
    {
      "chunk": "chunks/main-app-842b2bfec3a562ae.js",
      "size": 231,
    },
    {
      "chunk": "chunks/webpack-b7a1470890cdd00b.js",
      "size": 1832.96,
    },
  ],
  "sharedByAllTotal": 87347.2,
  "unit": "B",
}
`;
